packages/core/echo/echo-pipeline/src/db-host/query-service.ts

```diff
@@ -44,20 +45,35 @@ type ActiveQuery = {
 
 export class QueryServiceImpl extends Resource implements QueryService {
   private readonly _queries = new Set<ActiveQuery>();
+  private readonly _backpressureExecutor = new TimedTaskScheduler({
+    budget: 500,
+    budgetPeriod: 1000,
+    restTime: 500,
+    maxParallelTasks: 10,
+    cleanUpAfter: 30_000,
+  });
 
   private readonly _updateQueries = new DeferredTask(this._ctx, async () => {
-    await Promise.all(
-      Array.from(this._queries).map(async (query) => {
-        try {
-          const { changed } = await query.executor.execQuery();
-          if (changed) {
-            query.sendResults(query.executor.getResults());
+    if (this._queries.size === 0) {
+      return;
+    }
+
+    const tasks = Array.from(this._queries).map((query) => {
+      return {
+        run: async () => {
+          try {
+            const { changed } = await query.executor.execQuery();
+            if (changed) {
+              query.sendResults(query.executor.getResults());
+            }
+          } catch (err) {
+            log.catch(err);
           }
-        } catch (err) {
-          log.catch(err);
-        }
-      }),
-    );
+        },
+      };
+    });
+
+    await Promise.all(this._backpressureExecutor.schedule(tasks));
```

Comment:
is there any benifit in `schedule` taking an array of tasks? 
alternatively the code could look like:

```ts
await Promise.all(Array.from(this._queries).map(query => this._backpressureExecutor.schedule( async () => {
          try {
            const { changed } = await query.executor.execQuery();
            if (changed) {
              query.sendResults(query.executor.getResults());
            }
          } catch (err) {
            log.catch(err);
          }	          }
        } catch (err) {	        },)))